src/app/store/analytics/analytics.slice.ts

```diff
@@ -0,0 +1,32 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { ChainID } from '@stacks/common';
+
+interface AnalyticsState {
+  hadZeroStx: {
```

Comment:
I was curious on this too after thinking about my own action naming, do we prefer present or past tense?

Reply:
+1 to change it to `has` and use present tense

Reply:
I'm not exactly sure I understand the approach here per the [test cases](https://github.com/hirosystems/stacks-wallet-web/issues/2232#issuecomment-1040175260) in the related issue, which suggests only that there's a new "first_deposit" event that's fired for a given account whenever the balance goes from zero to non-zero for the first time (per chain data?).

If that's the case, why do we need an attribute like `hadZeroStx` (or similar) at all for (all) other events?

Incidentally, I'm not sure how the "first_deposit" approach is different than the "receive_asset" approach originally filed [here](https://github.com/hirosystems/stacks-wallet-web/issues/2034) aside from perhaps capturing fewer properties about the deposit (e.g. type and contract) / not supporting non-STX deposits?

Reply:
And per Fara's point, we do prefer present tense (e.g. "deposit_first_stx") over past tense or nouns (e.g. "first_deposit").

Reply:
@markmhx This hadZeroStx is an internal variable used to later fire the analytics event 'deposit_first_stx". This event is fired only when the user goes from 0 STX to a positive number.
Fara talked about redux actions not analytics even though 'deposit_first_stx' is a better name. So I will change the name of the event from 'first_deposit' to 'deposit_first_stx'.
"Also, wouldn't hadZeroStx apply to all accounts theoretically in your description ("the user had 0 STX at one point in the past") since all addresses start with no balance?"
hadZeroStx means our analytics has seen this state (it's under Analytics slice, not the account), so it knows when to trigger when it sees the first deposit.
Currently it works like this:
When the user creates an account or enters a seed with 0 STX, we set hadZeroStx to true.
Whenever the balance becomes positive, we set hadZeroStx to false and trigger the event. And everything works fine.
@kyranjamie using your initialStxDepositTxId it will be the same logic, except instead of storing a boolean indicating the presence of deposit, in your case we store the first deposit tx, whose value we don't really need. The problem with that is for older accounts, we don't have the first initialDepositTxId because we fetch only the last 50. The state will be inaccurate or you will have to do extra fetch.
How about just renaming 'hadZeroStx' to 'hasStxDeposits' and take the complement of the initial state and conditional check ?

Reply:
Renaming is fine, as long as it's clear 👍🏼